step-capstone/src/components/Trip.js

```diff
@@ -112,78 +21,59 @@ export default class Trip extends React.Component {
         this.handleAddItem = this.handleAddItem.bind(this);
     }
 
-    fetchData() {
-        // TODO: fetch from datastore
-        return testData;
-    }
-
     componentDidMount() {
-        let data = this.fetchData();
-        this.setState({
-            items: data
-        })
+      let travelObjectList = [];
+      this.context.getData(this.state.reference)
+      .then(data => {
+        data.data().travelObjects.forEach(travelObject => {
+          travelObjectList.push(travelObject)
+        });
+        this.setState({items : travelObjectList});
+      })
+      .catch(error => {console.log("Error Getting Trip Data")});
     }
 
-    handleRemoveItem(id) {
+    handleRemoveItem(data) {
+      this.context.deleteTravelObject(this.state.reference, data)
+      .then(() => {
         this.setState({
-            items: this.state.items.filter((item) => item.id !== id)
+          items: this.state.items.filter((item) => item.id !== data.id)
         });
+      })
+      .catch(error => console.log("Error Removing Item"));
     }
 
     handleEditItem(data) {
-        this.setState({
-            items: this.state.items.map((item) => {
-                if (item.id !== data.id) {
-                    return item;
-                } else {
-                    return data
-                }
-            })
-        })
+      this.setState({
+          items: this.state.items.map((item) => {
+              if (item.id === data.id) {
+                let success = this.context.editTravelObject(this.state.reference, item, data)
+                  .then(() => {
+                    return true;
+                  })
+                  .catch(error => {
+                    console.log("Error Editing Item");
+                    return false; 
+                  });
+                return (success ? data : item);
+              } else {
+                  return item;
+              }
+          })
+      })
     }
 
     handleAddItem(data) {
         if (data === undefined) {
-            console.log("please enter information")
+            console.log("please enter information");
```

Comment:
Should probably throw an error if this happens rather than a console.log(), because if it does something went wrong with the form (it's not supposed to allow creation of empty travel objects)

Reply:
i agree though I am not sure who would catch that error? and would we just want to display the form again?

Reply:
I don't think we'll encounter this error at all, since we'll only call handleAddItem when the user click Save and the handleSave() function will first check if the data is validated (if the data is empty or the dates are wrong...). Then, only when the data is validated, will the date be saved and the form popover be closed.
![image](https://user-images.githubusercontent.com/42331170/86633355-9001da80-bf9e-11ea-93d9-61624a5db230.png)

Reply:
ok I agree. I will just remove that section